Go/Go.sublime-syntax

```diff
@@ -416,44 +432,68 @@ contexts:
 
   match-floats:
     # Integer, no fraction, exponent
-    - match: \d+({{exponent}})\d+
+    - match: '{{ddigits}}{{exponent}}{{ddigits}}'
       scope: constant.numeric.float.go
-      captures:
-        1: punctuation.separator.exponent.go
     # Integer, fraction, optional exponent
-    - match: \d+(\.)\d+(?:({{exponent}})\d+)?
+    - match: '{{ddigits}}(\.){{ddigits}}(?:{{exponent}}{{ddigits}})?'
       scope: constant.numeric.float.go
       captures:
         1: punctuation.separator.decimal.go
-        2: punctuation.separator.exponent.go
     # Dot without fraction
     - match: \d+\.(?:{{exponent}}\d+)?
       scope: invalid.deprecated.go
     # Dot without integer
     - match: \.\d+(?:{{exponent}}\d+)?
       scope: invalid.deprecated.go
+    # Hex float literal, no fraction
+    - match: '(0[xX]){{hdigits}}[pP][+-]?{{ddigits}}(i)?'
+      scope: constant.numeric.float.go
+      captures:
+        1: punctuation.definition.numeric.hexadecimal.go
```

Comment:
Sorry for nagging, but would you please mind

1. to add the `[decimal|hexadecimal]` subscope to the `constant` scopes (here:  `constant.numeric.float.hexadecima.go`)
2. to use `punctuation.definition.numeric.base.go` as common scope for all the punctuation?

Reply:
Resolved in 8548792 and 3b30fb7.